modules/api/pkg/ee/clusterbackup/storage-location/handler.go

```diff
@@ -168,21 +193,31 @@ func CreateCBSL(ctx context.Context, request interface{}, provider provider.Back
 	}, nil
 }
 
-func DeleteCBSL(ctx context.Context, request interface{}, provider provider.BackupStorageProvider, projectProvider provider.ProjectProvider) error {
+func DeleteCBSL(ctx context.Context, request interface{}, userInfoGetter provider.UserInfoGetter, provider provider.BackupStorageProvider, projectProvider provider.ProjectProvider) error {
 	req, ok := request.(deleteCbslReq)
 	if !ok {
 		return utilerrors.NewBadRequest("invalid request")
 	}
 
-	err := provider.DeleteUnsecured(ctx, req.ClusterBackupStorageLocationName)
+	user, err := userInfoGetter(ctx, req.ProjectID)
+
+	if err != nil {
+		return err
+	}
+
+	if user.Roles.Has("viewers") {
+		return fmt.Errorf("user with a viewer role is not permitted to delete storage locations.")
+	}
```

Comment:
my understanding from other code is that it is possible for a user to have both "viewer" as well as another role. As a result we also need to check if viewers is the only role the user has.

For example like this is done here
https://github.com/kubermatic/dashboard/blob/140d3a4b773b36c18b9669d86fd7f348cdfd388b/modules/api/pkg/provider/kubernetes/cluster.go#L428

Reply:
Same for Create and Patch as well